### ISSUE TITLE
[BUG] Type with a member that has initializer and defined operator=: (out this, n : std::name ) produces an assignment operator that badly assign member with initializer

### ISSUE THREAD

**Issue by filipsajdak** [hsutter/cppfront#291]
In the current implementation of cppfront (https://github.com/hsutter/cppfront/commit/fbf55ad9efcded58f550f05ede37b5b48399a7fd) the following code:

```cpp
element: type = {
    children: std::vector<int> = ();
    name: std::string;

    operator=: (out this, n : std::string ) = {
        name = n;
    }
}
```
Generates:
```cpp


#include "cpp2util.h"


#line 1 "/Users/filipsajdak/dev/execspec/external/tests/bug_assignement_operator_2.cpp2"
class element;

//=== Cpp2 definitions ==========================================================

#line 1 "/Users/filipsajdak/dev/execspec/external/tests/bug_assignement_operator_2.cpp2"
class element  {
    private: std::vector<int> children {}; 
    private: std::string name; 

    public: explicit element(cpp2::in<std::string> n)
        : name{ n }
#line 6 "/Users/filipsajdak/dev/execspec/external/tests/bug_assignement_operator_2.cpp2"
{
    }
#line 5 "/Users/filipsajdak/dev/execspec/external/tests/bug_assignement_operator_2.cpp2"
    public: auto operator=(cpp2::in<std::string> n) -> element& {
        children = ;
        name = n;
#line 6 "/Users/filipsajdak/dev/execspec/external/tests/bug_assignement_operator_2.cpp2"
        
        return *this;
#line 7 "/Users/filipsajdak/dev/execspec/external/tests/bug_assignement_operator_2.cpp2"
    }
};
```
The issue is that the assignment operator has one assignment to the `children` variable that is not complete:
```cpp
        children = ;
```

**Comment by filipsajdak** [comment]
I found a workaround and resigned from having a default initializer for my members. The following code:

```cpp
element: type = {
    children: std::vector<int>;
    name: std::string;

    operator=: (out this, n : std::string ) = {
        children = std::vector<int>(); // it would be good to be able to spell it: children = ();
        name = n;
    }
}
```
works and generate:
```cpp
#include "cpp2util.h"

#line 1 "/Users/filipsajdak/dev/execspec/external/tests/bug_assignement_operator_2.cpp2"
class element;

//=== Cpp2 definitions ==========================================================

#line 1 "/Users/filipsajdak/dev/execspec/external/tests/bug_assignement_operator_2.cpp2"
class element  {
    private: std::vector<int> children; 
    private: std::string name; 

    public: explicit element(cpp2::in<std::string> n)
        : children{ std::vector<int>() }
        , name{ n }
#line 5 "/Users/filipsajdak/dev/execspec/external/tests/bug_assignement_operator_2.cpp2"
                                            {

#line 8 "/Users/filipsajdak/dev/execspec/external/tests/bug_assignement_operator_2.cpp2"
    }
#line 5 "/Users/filipsajdak/dev/execspec/external/tests/bug_assignement_operator_2.cpp2"
    public: auto operator=(cpp2::in<std::string> n) -> element& {
        children = std::vector<int>();
        name = n;
        return *this;
#line 8 "/Users/filipsajdak/dev/execspec/external/tests/bug_assignement_operator_2.cpp2"
    }
};
```

**Comment by hsutter** [comment]
Thanks. Like #290, I think this is fixed by feff6407b16298118003359638a08c85ad0d61ee. Please check, including the commit warning -- see #290 comment thread.

**Comment by filipsajdak** [comment]
@hsutter I confirmed it is solved!
